packages/store/addon/-private/system/model/shim-model-class.ts

```diff
@@ -0,0 +1,63 @@
+import { map } from '@ember/object/computed';
+import CoreStore from '../core-store';
+import { stringify } from 'querystring';
+import { RelationshipSchema, AttributeSchema } from '../../ts-interfaces/record-data-schemas';
+import { Dict } from '../../ts-interfaces/utils';
+
+// Mimics the static apis of DSModel
+export default class ShimModelClass {
+  store: CoreStore;
+  modelName: string;
+  _id?: string;
+
+  // TODO Maybe expose the class here?
+  constructor(store: CoreStore, modelName: string, id?: string) {
+    this.store = store;
+    this.modelName = modelName;
+    this._id = id;
+  }
+
+  get fields(): Map<string, 'attribute' | 'belongsTo' | 'hasMany'> {
```

Comment:
we should fast follow a deprecation, I believe as part of the RFC we were looking at the shim being a place to do. Seems good to have full API support at the beginning though.

Reply:
:+1: